web/client/codechecker_client/credential_manager.py

```diff
@@ -27,6 +28,33 @@
 SESSION_COOKIE_NAME = _SCN
 
 
+def simplify_credentials(credentials):
+    """ Replace the key of the entries with host:port values.
+
+    Returns a new dictionary from the credentials where the key will be
+    replaced by 'host:port' values so protocols and product names will be
+    removed if these are given.
+    """
+    host_entry_pattern = re.compile(
```

Comment:
Wouldn't be easier to use the `urlparse.urlsplit` to split up the url?

Reply:
I checked it and it works differently on different parameters and it is hard to decide which field we need to use and we also have to post-process the result.

```py
urlparse.urlsplit("my.server.com:443/Test")
# SplitResult(scheme='my.server.com', netloc='', path='443/Test', query='', fragment='')

urlparse.urlsplit("my.server.com:443")
SplitResult(scheme='', netloc='', path='my.server.com:443', query='', fragment='')

urlparse.urlsplit("*:8080")
# SplitResult(scheme='', netloc='', path='*:8080', query='', fragment='')

urlparse.urlsplit("https://my.server.com:443/Test")
# SplitResult(scheme='https', netloc='my.server.com:443', path='/Test', query='', fragment='')
```